test/auto_parallel/spmd_rules/test_c_embedding_rule.py

```diff
@@ -0,0 +1,165 @@
+
+    def test_c_embedding_infer_forward(self):
+        # forward setup
+        table_shape = [512, 768]  # [V,H]
+        x_shape = [4, 1024]  # [B,S]
+        process_mesh = auto.ProcessMesh(mesh=[[0, 1, 2, 3], [4, 5, 6, 7]])
+
+        table_tensor_dist_attr = TensorDistAttr()
+        table_tensor_dist_attr.process_mesh = process_mesh
+        self.table_dist_tensor_spec = DistTensorSpec(
+            table_shape, table_tensor_dist_attr
+        )
+
+        x_tensor_dist_attr = TensorDistAttr()
+        x_tensor_dist_attr.process_mesh = process_mesh
+        self.x_dist_tensor_spec = DistTensorSpec(x_shape, x_tensor_dist_attr)
+
+        self.attrs = OrderedDict([('start_index', 0), ('vocab_size', -1)])
+
+        # data parallel
+        self.table_dist_tensor_spec.set_dims_mapping([-1, -1])
+        self.x_dist_tensor_spec.set_dims_mapping([1, -1])
+        result_dist_attrs = self.rule.infer_forward(
+            self.table_dist_tensor_spec,
+            self.x_dist_tensor_spec,
+            self.attrs['start_index'],
+            self.attrs['vocab_size'],
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
+        self.assertEqual(len(result_dist_attrs), 2)
+        self.assertEqual(len(infered_input_dist_attrs), 2)
+        self.assertEqual(len(infered_output_dist_attrs), 1)
+
+        self.assertEqual(infered_input_dist_attrs[0].dims_mapping, [-1, -1])
+        self.assertEqual(infered_input_dist_attrs[1].dims_mapping, [1, -1])
+        self.assertEqual(infered_output_dist_attrs[0].dims_mapping, [1, -1, -1])
+
+        # table row-wise parallel
+        self.table_dist_tensor_spec.set_dims_mapping([1, -1])
+        self.x_dist_tensor_spec.set_dims_mapping([-1, -1])
+        result_dist_attrs = self.rule.infer_forward(
+            self.table_dist_tensor_spec,
+            self.x_dist_tensor_spec,
+            self.attrs['start_index'],
+            self.attrs['vocab_size'],
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
+        self.assertEqual(infered_input_dist_attrs[0].dims_mapping, [1, -1])
+        self.assertEqual(infered_input_dist_attrs[1].dims_mapping, [-1, -1])
+        self.assertEqual(
+            infered_output_dist_attrs[0].dims_mapping, [-1, -1, -1]
+        )
+        self.assertEqual(infered_output_dist_attrs[0]._is_partial(), True)
+        self.assertEqual(infered_output_dist_attrs[0]._partial_dims(), {1})
+
+    def test_c_embedding_infer_backward(self):
+        # backward setup
+        process_mesh = auto.ProcessMesh(mesh=[[0, 1, 2, 3], [4, 5, 6, 7]])
+
+        table_shape = [512, 768]  # [V,H]
+        x_shape = [4, 1024]  # [B,S]
+
+        table_tensor_dist_attr = TensorDistAttr()
+        table_tensor_dist_attr.process_mesh = (
+            process_mesh  # not set the dims mapping is ok.
+        )
```

Comment:
`table_tensor_dist_attr.process_mesh = process_mesh`

Reply:
Thx, the code has been modified according to the suggestions above.